eureka/niriss_class.py

```diff
@@ -0,0 +1,516 @@
+import os
+import numpy as np
+from astropy import units
+from astropy.io import fits
+import matplotlib.pyplot as plt
+from astropy.table import Table
+
+from .lib.tracing_niriss import mask_method_edges, mask_method_ears, ref_file
+from .lib.masking        import (interpolating_row, data_quality_mask,
+                              interpolating_image)
+from .lib.clipping       import time_removal
+from .S3_data_reduction.background     import bkg_sub, fitbg3
+from .S3_data_reduction.niriss_extraction   import (dirty_mask, box_extract,
+                                              optimal_extraction_routine)
+#from .S3_data_reduction.niriss  import wave_NIRISS as wavelength
+from .lib.simultaneous_order_fitting import fit_orders, fit_orders_fast
+
+
+__all__ = ['NIRISS_S3']
+
+
+class NIRISS_S3(object):
+
+    def __init__(self, filename, f277_filename=None,
+                 data_dir=None, output_dir=None):
+        """
+        Initializes the NIRISS S3 data reduction class.
+
+        Parameters
+        ----------
+        filename : str
+           The name of the FITS file with NIRISS observations.
+        data_path : str, optional
+           The path to where the input FITS files are stored. Default
+           is None. If None, will search the current working directory
+           for the files.
+        output_dir : str, optional
+           The path where output files will be saved. Default is None.
+           if None, will save all files to the current working directory.
+
+        Attributes
+        ----------
+        filename : str
+           The science file name.
+        data_path : str
+           The path to where the science file is stored.
+        output_dir : str
+           The path to where output files will be stored.
+        """
+        self.filename = filename
+
+        if data_dir is not None:
+            self.data_dir = data_dir
+        else:
+            self.data_dir = os.getcwd()
+
+        if output_dir is not None:
+            self.output_dir = output_dir
+        else:
+            self.output_dir = os.getcwd()
+
+        # Opens the science FITS file and sets up all proper
+        #   data attributes for the reduction steps.
+        self.setup()
+
+        self.dq_mask = data_quality_mask(self.dq)
+
+        if f277_filename is not None:
+            self.setup_f277(f277_filename)
+        else:
+            self.f277 = None
+            print('Without F277W filter image, some functions may not be available.')
+
+        self.trace_ear     = None
+        self.trace_edge     = None
+        self.bkg      = None
+        self.box_mask = None
+        self.box_var1 = None
+        self.box_var2 = None
+        self.bkg_removed  = None
+        self.box_spectra1 = None
+        self.box_spectra2 = None
+        self.box_mask_separate = None
+
+        return
+
+
+    def setup(self):
+        """
+        Sets up all proper attributes from the FITS file.
+
+        Attributes
```

Comment:
The attributes of a class should be documented up above, before the init function